script.js

```diff
@@ -1,4 +1,254 @@
+      }
+    }
+
+    //game rules - how to lose
+
+    if (
+      (input == "paper" && program == "scissors") ||
+      (input == "stone" && program == "paper") ||
+      (input == "scissors" && program == "stone")
+    ) {
+      userWon = userWon;
+      computerWon = computerWon + 1;
+      draw = draw;
+      percentageWon = (userWon / turnsPlayed) * 100;
+
+      if (percentageWon >= 50) {
+        return (myOutputValue = `You chose ${inputObject} <br> The computer chose ${outputObject}. <br><br>You lose! Bummer.<br><br>So far ${userName}, you have been winning ${userWon}/${turnsPlayed} turns (or ${percentageWon}%).Still doing well!`);
+      } else if (percentageWon < 50) {
+        return (myOutputValue = `You chose ${inputObject} <br> The computer chose ${outputObject}. <br>You lose! Bummer.<br><br>So far ${userName}, you have been winning ${userWon}/${turnsPlayed} turns (or ${percentageWon}%). Hope your luck turns!`);
+      }
+    }
+
+    //game rules - how to draw
+    if (
+      (input == "paper" && program == "paper") ||
+      (input == "stone" && program == "stone") ||
+      (input == "scissors" && program == "scissors")
+    ) {
+      userWon = userWon;
+      computerWon = computerWon;
+      draw = draw + 1;
+      percentageWon = (userWon / turnsPlayed) * 100;
+
+      if (percentageWon >= 50) {
+        return (myOutputValue = `You chose ${inputObject} <br> The computer chose ${outputObject}. <br><br>It's a draw!<br><br>So far ${userName}, you have been winning ${userWon}/${turnsPlayed} turns (or ${percentageWon}%). Still doing well!`);
+      } else if (percentageWon < 50) {
+        return (myOutputValue = `You chose ${inputObject} <br> The computer chose ${outputObject}. <br>It's a draw!<br><br>So far ${userName}, you have been winning ${userWon}/${turnsPlayed} turns (or ${percentageWon}%). Hope your luck turns!`);
+      }
+    }
+
+    if (input == "reverse") {
+      turnsPlayed = turnsPlayed - 1;
+      currentGameMode = "reverse SPS";
+      return (myOutputValue =
+        "You are now playing reverse scissors-paper-stone!");
+    }
+
+    //input validation: if user doees not input any of three valid options, system should advise on correct input options.
+
+    myOutputValue = `Our program does not recognise your input "${input}". <br> Please try again by entering one of the following: scissors, paper, or stone.`;
+  } else if (currentGameMode == "reverse SPS") {
+    //format the input object name
+    var inputObjectFormatted = function (input) {
+      if (input == "scissors") {
+        return "scissors ✂️";
+      }
+      if (input == "paper") {
+        return "paper 🗒";
+      }
+      if (input == "stone") {
+        return "stone 🥌";
+      }
+    };
```

Comment:
there are a lot of if-if else pairs in the code

if()
else if ()


if()
else if ()


if()
else if ()

It would be easier to read and track if it was more standardised, this would give you a similar outcome

if()
else if()
else if()
else if()
else()

this would do too

if()
if()
if()
if()